Modules/Core/Common/include/itkImageBase.h

```diff
@@ -768,6 +768,19 @@ class ITK_TEMPLATE_EXPORT ImageBase : public DataObject
   bool
   VerifyRequestedRegion() override;
 
+  /** Check whether this image and the other image occupy the same physical space.
```

Comment:
May be something like this:
"Checks whether the images' pixels with the same index occupy the same physical space.

Compares the origin, spacing, and direction for equality within provided tolerances. There is no
check for valid regions in between the images."